tests/operator/helper.go

```diff
@@ -45,3 +57,62 @@ func SplitCsv(csv string) CsvResult {
 	}
 	return result
 }
+
+func hasOperatorInstallModeSingleNamespace(installModes []v1alpha1.InstallMode) bool {
+	for i := 0; i < len(installModes); i++ {
+		if installModes[i].Type == v1alpha1.InstallModeTypeSingleNamespace && installModes[i].Supported {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSingleNamespacedOperatorUnderTest(operators []*provider.Operator) (singleNamespacedOperators []*provider.Operator) {
+	for _, operator := range operators {
+		if hasOperatorInstallModeSingleNamespace(operator.Csv.Spec.InstallModes) && len(operator.TargetNamespaces) == 1 {
+			singleNamespacedOperators = append(singleNamespacedOperators, operator)
+		}
+	}
+	return singleNamespacedOperators
+}
+
+// This function checks if the namespace contains only valid operator pods
+func checkIfNamespaceContainsOnlyOperatorPods(namespace string) (isOperatorOnlyNamespace bool, err error) {
+	// Get all pods from the target namespace
+	podsList, err := clientsholder.GetClientsHolder().K8sClient.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return isOperatorOnlyNamespace, err
+	}
+	foundCsvs := make(map[string]bool)
+	foundOperatorPods := 0
+	for index := range podsList.Items {
+		// Get the top owners of the pod
+		pod := podsList.Items[index]
+		topOwners, err := podhelper.GetPodTopOwner(pod.Namespace, pod.OwnerReferences)
+		if err != nil {
+			return isOperatorOnlyNamespace, fmt.Errorf("could not get top owners of Pod %s (in namespace %s), err=%v", pod.Name, pod.Namespace, err)
+		}
+
+		// check if owner matches with the csv
+		for _, owner := range topOwners {
+			// The owner must be in the targetNamespace
+			if owner.Kind == v1alpha1.ClusterServiceVersionKind && owner.Namespace == namespace {
+				foundOperatorPods++
+				foundCsvs[owner.Name] = true
+				break
+			}
+		}
+	}
+
+	// Check if the found CSVs contain only valid operators under test
+	status := false
```

Comment:
`status` seems to be true if one operator matches only one found csv and is not clusterwide. Instead, all found operators that are under test should not be clusterwide, something like:


```
    AllOperatorsFoundInNamespaceAreValid := true
    // report an error only if an operator under test is found to be clusterwide
     for _, operator := range env.Operators {
          for foundCsv := range foundCsvs {
			if operator.IsClusterWide && operator.Csv.Name == foundCsv {
				AllOperatorsFoundInNamespaceAreValid = false
                                  break
			}
		}
	}
```